### ISSUE TITLE
[$1000] Chat - Emojis in bold and italic style appear smaller in size

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#15424]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go to URL https://staging.new.expensify.com/
2. Login with any account
3. Go to any chat
4. Write some emojis in bold and italic style

## Expected Result:
Emojis in bold and italic style appear in same size

## Actual Result:
Emojis in bold and italic style appear smaller in size

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.2.76.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![2302](https://user-images.githubusercontent.com/93399543/221019027-c0fd6bdf-12f7-442c-8833-32d9484be423.png)

https://user-images.githubusercontent.com/93399543/221019030-0136f649-a06b-41d8-ac76-a303f1b51e37.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01581ef3f6d9cefd64</li>
        <li>Upwork Job ID: 1630355063793573888</li>
        <li>Last Price Increase: 2023-02-27</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @bfitzexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01581ef3f6d9cefd64

**Comment by MelvinBot** [comment]
Triggered auto assignment to @trjExpensify (`External`), see https://stackoverflow.com/c/expensify/questions/8582 for more details.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @eVoloshchak (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @arosiclair (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by trjExpensify** [comment]
Errr.. @bfitzexpensify why did this add me as well?

**Comment by bfitzexpensify** [comment]
Not sure @trjExpensify but I can handle

**Comment by bernhardoj** [comment]
There is an [issue](https://github.com/Expensify/App/issues/4114) that will show the emoji larger based on the text. I already asked [here](https://github.com/Expensify/App/issues/4114#issuecomment-1447484588) whether it will also handle the emojis inside a markdown.

**Comment by jamesarosen** [comment]
Some additional notes on current behavior:

* `🌻` on its own line appears large
* `*🌻*` on its own line appears normal text size
* `_🌻_` on its own line appears normal text size, slanted
* `foo: 🌻` appears normal size
* `A plain emoji 🌻. A bold emoji: *🌻*. An italic emoji: _🌻_` -- all appear the same size; the last is slanted

**Comment by MelvinBot** [comment]
📣 @jamesarosen! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by kaushiktd** [comment]
- Please re-state the problem that we are trying to solve in this issue.
    Chat - Emojis in bold and italic style appear smaller in size

- What is the root cause of that problem?
    When sending emojis with the italic or bold format, it will add HTML tags to 
    selected emojis, so it will show as small in size.

- What changes do you think we should make in order to solve the problem?
    We need to add font size to HTML tags for bold and italic in below mentioned file:
   https://github.com/Expensify/App/blob/main/src/pages/home/report/ReportActionItemFragment.js#L113

**Comment by mazr422** [comment]
- Chat emojis in bold and italic style appear smaller in size.
I think this issue is not stated correctly. it should be **_Chat emojis appear larger in size, emojis in bold and italic style appear in right size._**


What is the root cause of that problem?
What changes do you think we should make in order to solve the problem?
What alternative solutions did you explore? (Optional)
Reminder: Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by MelvinBot** [comment]
📣 @mazr422! 📣

Hey, it seems we don’t have your contributor details yet! You'll only have to do this once, and this is how we'll hire you on Upwork.
Please follow these steps:

1. Get the email address used to login to your Expensify account. If you don't already have an Expensify account, create one [here](https://new.expensify.com/). If you have multiple accounts (e.g. one for testing), please use your main account email.
2. Get the link to your Upwork profile. It's necessary because we only pay via Upwork. You can access it by logging in, and then clicking on your name. It'll look like [this](https://www.upwork.com/freelancers/~01ca8ad8ce36b9eceb). If you don't already have an account, sign up for one [here](https://www.upwork.com/nx/signup/?dest=home).
3. Copy the format below and paste it in a comment on this issue. Replace the placeholder text with your actual details.

![Screen Shot 2022-11-16 at 4 42 54 PM](https://user-images.githubusercontent.com/30609178/202326034-1a292e64-48fd-4fef-9f4f-7feaaeedfd9f.png)

Format:
```
Contributor details
Your Expensify account email: <REPLACE EMAIL HERE>
Upwork Profile Link: <REPLACE LINK HERE>
```

**Comment by alexxxwork** [comment]
@arosiclair @eVoloshchak @bfitzexpensify 
I believe this issue should be closed or placed on hold because there's another issue with solution when emojis shouldn't be bold or italic #14676

**Comment by arosiclair** [comment]
> @arosiclair @eVoloshchak @bfitzexpensify I believe this issue should be closed or placed on hold because there's another issue with a solution where emojis shouldn't be bold or italic #14676

Agreed. Bolding/italicizing emoji's is the first problem to solve. Let's hold on that issue and then re-test to see if sizing is still a problem afterwards

**Comment by bfitzexpensify** [comment]
Cool, agreed. I've subscribed to https://github.com/Expensify/App/issues/14676 and switched this to weekly. Once https://github.com/Expensify/App/issues/14676 is resolved, will check back in here.

**Comment by arosiclair** [comment]
Still on hold

**Comment by bfitzexpensify** [comment]
Still on hold

**Comment by arosiclair** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Still on hold

**Comment by bfitzexpensify** [comment]
Still on hold

**Comment by arosiclair** [comment]
Maybe someday

**Comment by bfitzexpensify** [comment]
https://github.com/Expensify/App/issues/14676 is also on hold and monthly, so going to change this to the same priority

**Comment by arosiclair** [comment]
Still held

**Comment by arosiclair** [comment]
This is still held on #14676 which is held on https://github.com/facebook/hermes/pull/968. It looks like that PR may be [completed soon and included in **v0.73** of hermes](https://github.com/facebook/hermes/pull/968#issuecomment-1633011552).

**Comment by arosiclair** [comment]
Same as before

**Comment by bfitzexpensify** [comment]
Still held

**Comment by bfitzexpensify** [comment]
Still held

**Comment by arosiclair** [comment]
Looks like we are now held on https://github.com/Expensify/App/pull/31558

**Comment by bfitzexpensify** [comment]
Still held on https://github.com/Expensify/App/pull/31558

**Comment by arosiclair** [comment]
It looks like #31558 is merged so this is should be unblocked now 😮. Tbh I have no idea where this issue stands at this point. I think we need to re-test according to my old comment:

> Agreed. Bolding/italicizing emoji's is the first problem to solve. Let's hold on that issue and then re-test to see if sizing is still a problem afterwards

@eVoloshchak do you mind doing that?

**Comment by bfitzexpensify** [comment]
Yeah, I can no longer bold or italicize emojis at all either on prod or staging, so the issue here doesn't exist either. I think we're good to close this out.
